src/detector/implementations/aruco_marker_implementation.py

```diff
@@ -0,0 +1,280 @@
+from src.detector.api import \
+    GetDetectionParametersResponse, \
+    GetMarkerSnapshotsRequest, \
+    GetMarkerSnapshotsResponse, \
+    SetDetectionParametersRequest
+from src.common import \
+    EmptyResponse, \
+    ErrorResponse, \
+    get_kwarg
+from src.common.structures import \
+    CornerRefinementMethod, \
+    CORNER_REFINEMENT_METHOD_DICTIONARY_INT_TO_TEXT, \
+    CORNER_REFINEMENT_METHOD_DICTIONARY_TEXT_TO_INT, \
+    DetectionParameters, \
+    MarkerSnapshot, \
+    MarkerCornerImagePoint
+
+from src.common.structures.marker_status import MarkerStatus
+
+import datetime
+import logging
+from typing import Any, Callable
+import cv2.aruco
+import numpy
+
+from src.detector.implementations import abstract_marker_interface
+
+logger = logging.getLogger(__name__)
+
+class ArucoMarker(abstract_marker_interface.AbstractMarkerInterface):
+    _marker_status: MarkerStatus  # internal bookkeeping
+
+    _marker_dictionary: Any | None  # created by OpenCV, type cv2.aruco.Dictionary
+    _marker_parameters: Any  # created by OpenCV, type cv2.aruco.DetectorParameters
+    _marker_label_reverse_dictionary: dict[int, str]
+    _marker_detected_snapshots: list[MarkerSnapshot]
+    _marker_timestamp_utc: datetime.datetime
+
+    def __init__(
+            self
+    ):
+        # TODO: Maybe these are best left at None to begin
```

Comment:
This is probably just copy-pasted. Feel free to remove it, or change it to say that the values are currently hardcoded but may be made changeable at some point in the future.